lib/Dashboard.js

```diff
@@ -0,0 +1,54 @@
+var rp = require('./request-promise'),
+  config = require('../config'),
+  _ = require('lodash'),
+  Q = require('q');
+
+var Module = require('./Module');
+
+function Dashboard (slug) {
+  this.options = {
+    json: true,
+    slug: slug
+  };
+}
+
+Dashboard.prototype.getConfig = function () {
+  var options = _.extend(this.options, {
+    url: config.stagecraft + 'public/dashboards?slug=' + this.options.slug
+  });
+
+  return rp(options);
+};
+
+Dashboard.prototype.resolve = function () {
```

Comment:
Agreed i don't know if resolve() could gain a shallow and a deep version ie:

`dashboard.resolve(true)` would recurse down. That doesn't read the best but maybe it can be a constant or string or something.